lightning/src/ln/channelmonitor.rs

```diff
@@ -1245,6 +1241,21 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 				self.their_cur_revocation_points = Some((commitment_number, their_revocation_point, None));
 			}
 		}
+		self.payment_keys = match self.key_storage {
+			Storage::Local { ref payment_base_key, .. } => {
+				if let Ok(local_key) = chan_utils::derive_private_key(&self.secp_ctx, &their_revocation_point, &payment_base_key) {
+					let local_payment_p2wpkh = {
+						// Note that the Network here is ignored as we immediately drop the address for the
+						// script_pubkey version.
+						let payment_hash160 = Hash160::hash(&PublicKey::from_secret_key(&self.secp_ctx, &local_key).serialize());
```

Comment:
Oof. Please refactor to avoid hashing in the per-tx loop. The previous version checked the input txid first, which seems reasonable to continue doing here.

Reply:
Thanks for the catch, in fact that's useless because we may detect first when a remote commitment tx is onchain and we may have a payment pending.